src/string.cr

```diff
@@ -3314,7 +3314,7 @@ class String
           if mem
             if char == '_'
               # case 3
-            else
+            elsif upcase || downcase
```

Comment:
I'm not really sure of how "123ABC" should be converted (if "ABC123" has to be converted as "abc123", why make a difference for "123ABC" ?).

The best to do for now is probably to wait for an answer of the core devs ?